app/src/main/java/com/example/drawingApp/editLayerAdapter.kt

```diff
@@ -0,0 +1,77 @@
+package com.example.drawingApp
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+
+class EditLayerAdapter(
+    private val data: List<LayerViewModel>,
+    /**
+     * called when the edit image is clicked in each view of the recycler view
+     * determines what will be performed when clicked
+     * currently allows the editing of the layer at that position of data
+     */
+    private val onEditClick: (LayerViewModel) -> Unit,
+    /**
+     * called when the delete image is clicked in each view of the recycler view
+     * determines what will be performed when clicked
+     * currently deletes the layer at that position of data
+     */
+    private val onDeleteClick: (LayerViewModel) -> Unit
+) : RecyclerView.Adapter<ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        return LayerViewHolder(
+            inflater.inflate(R.layout.edit_layer_dialog, parent, false),
+            onEditClick,
+            onDeleteClick
+        )
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(data[position])
+    }
+
+    override fun getItemCount(): Int = data.size
+
+}
+
+class LayerViewModel(val text: String)
```

Comment:
Along with my previous comment, I think you need a `Layer` object for the concept of a layer, but here this makes sense to leave as `LayerViewModel` because this is merely _presenting_ rather than doing Layer logic, so this is fine to keep, but make a `Layer` object for uses outside the RecyclerView

Reply:
Please revisit, on the next commit, I think a Layer object will definitely be needed down the line, but I changed back the layerNames from layerViewModels back to a list of strings.